pyat/at/physics/dynamic_aperture.py

```diff
@@ -0,0 +1,895 @@
+
+    dict_def_range = {'x': [-3e-2, 3e-2],
+                      'xp': [-1e-3, 1e-3],
+                      'y': [-1e-2, 1e-2],
+                      'yp': [-1e-3, 1e-3],
+                      'delta': [-2e-1, 2e-1],
+                      'ct': [-1e-1, 1e-1],
+                      }
+    """
+    @property
+    def dict_def_range(self):
+        return self._dict_def_range
+
+    @dict_def_range.setter
+    def dict_def_range(self, new_value):
+        self._dict_def_range = new_value
+        self.test_points_grid()  # recompute grid points if ranges are changed
+    """
+
+    def __init__(self,
+                 ring,
+                 dpp=0.0,
+                 mode=None,
+                 start_index=0,
+                 grid_mode='grid',
+                 compute_range=False,
+                 n_turns=2**10,
+                 search_divider=10,
+                 use_radiation=True
+                 ):
+        """
+        Acceptance6D computes the 6D acceptance for a pyAT lattice
+
+        :param ring: pyAT lattice
+        :param dpp: momentum deviation
+        :param mode: mode for computation. None = 6D (default), 'x-y', 'delta-x'. 'x-xp',...
+        :param start_index: index in ring where to start the computation
+        :param grid_mode: 'grid' or 'radial' set of test points to compute Acceptance
+        :param compute_range: compute range for each plane with more than 1 point to scan
+        :param nturns: number of turns that a particle must survive
+        :param search_divider: division of range used by recursive range
+        :param use_radiation: default true (turn on radiation, in future version, use radiation state of input lattice)
+        : n_point: dictionary to determine the # points to scan in each dimension
+        : dict_def_range: default range for each dimension
+        : dict_unit:  default units for each dimension
+
+        """
+
+        self.parallel_computation = True
+
+        # rotate lattice to given element
+        self.ring = at.Lattice(ring[start_index:] + ring[:start_index])
+
+        self.dpp = dpp
+        self.mode = mode
+
+        use_radiation = ring.radiation
+
+        self.ring.radiation_off()
+
+        # if self.dpp != 0.0:
+        self.orbit = self.ring.find_orbit4(dp=self.dpp)  # dpp is added to orbit here
+        #else:
+        #    self.orbit = self.ring.find_orbit6()
+
+        if use_radiation:
+            self.ring.radiation_on()
+        else:
+            self.ring.radiation_off()
+
+        self.coordinates = {'x': [],
+                            'xp': [],
+                            'y': [],
+                            'yp': [],
+                            'delta': [],
+                            'ct': []
+                            }
+        self.survived = []
+
+        self.number_of_turns = n_turns
+        self.search_divider = search_divider
+
+        self.grid_modes = ['grid', 'radial']
+        self.grid_mode = self.grid_modes[0]
```

Comment:
Apparently useless: self.grid_mode is positioned in `test_points_grid` called at the end.

Reply:
the grid_mode is used also by the function dynamic_aperture and Acceptance6D.get_border.
I have made a more consistent use of it. it was somewhere a function input and somewhere a class property.